state/azure/cosmosdb/cosmosdb.go

```diff
@@ -156,79 +155,39 @@ func (c *StateStore) Init(meta state.Metadata) error {
 	}
 	config.WithAppIdentifier("dapr-" + logger.DaprVersion)
 
+	c.client = documentdb.New(m.URL, config)
+	c.metadata = m
+	c.contentType = m.ContentType
+
 	// Retries initializing the client if a TooManyRequests error is encountered
 	bo := backoff.NewExponentialBackOff()
 	bo.InitialInterval = 2 * time.Second
 	bo.MaxElapsedTime = 5 * time.Minute
-	err = backoff.RetryNotify(func() (err error) {
-		client := documentdb.New(m.URL, config)
-
-		dbs, err := client.QueryDatabases(&documentdb.Query{
-			Query: "SELECT * FROM ROOT r WHERE r.id=@id",
-			Parameters: []documentdb.Parameter{
-				{Name: "@id", Value: m.Database},
-			},
-		})
-
-		if err != nil {
-			if isTooManyRequestsError(err) {
-				return err
-			}
-			return backoff.Permanent(err)
-		} else if len(dbs) == 0 {
-			return backoff.Permanent(fmt.Errorf("database %s for CosmosDB state store not found", m.Database))
-		}
-
-		c.db = &dbs[0]
-		colls, err := client.QueryCollections(c.db.Self, &documentdb.Query{
-			Query: "SELECT * FROM ROOT r WHERE r.id=@id",
-			Parameters: []documentdb.Parameter{
-				{Name: "@id", Value: m.Collection},
-			},
-		})
-		if err != nil {
-			if isTooManyRequestsError(err) {
-				return err
-			}
-			return backoff.Permanent(err)
-		} else if len(colls) == 0 {
-			return backoff.Permanent(fmt.Errorf("collection %s for CosmosDB state store not found.  This must be created before Dapr uses it", m.Collection))
-		}
-
-		c.metadata = m
-		c.collection = &colls[0]
-		c.client = client
-		c.contentType = m.ContentType
-
-		sps, err := c.client.ReadStoredProcedures(c.collection.Self)
-		if err != nil {
-			if isTooManyRequestsError(err) {
-				return err
-			}
-			return backoff.Permanent(err)
-		}
-
-		// get a link to the sp
-		for i := range sps {
-			if sps[i].Id == storedProcedureName {
-				c.sp = &sps[i]
-
-				break
+	err = backoff.RetryNotify(func() (innerErr error) {
+		_, innerErr = c.findCollection()
```

Comment:
Because of the latest update in the upstream SDK that allows us to address collections and sprocs via their ID and not just a self link, I've simplified this to a single call to `ReadCollection` rather than 2 queries before. This is also a simpler query that runs faster and uses less RUs :)